jme3-alloc/src/main/java/com/jme3/alloc/util/NativeBufferUtils.java

```diff
@@ -24,77 +55,69 @@ private NativeBufferUtils() {
-     * @return a new java-nio buffer object
+     * @param size the size of the newly created buffer in bytes 
+     * @return a newly created un-initialized {@link java.nio.ByteBuffer} object
+     * @see NativeBufferUtils#clearAlloc(long)
      */
-    public static native Buffer memoryAlloc(final long size);
+    public static native ByteBuffer memoryAlloc(final long size);
 
     /**
-     * Copies the value [value] into each of the first [size] bytes of the object beginning at the 
-     * memory address of [buffer].
+     * Allocates a buffer block of [size] bytes using [void* calloc(size_t)] provided by GNU stdlib, and initializes its elements to zero.
+     * Note: Maximum size is 32-bit for 32-bit architectures and 64-bit for 64-bit architectures.
      * 
-     * @param buffer a buffer block to start from
-     * @param value the value to set the memory blocks with starting from the buffer
-     * @param size the size to extend after the buffer address
-     * @return the buffer block
+     * @param size the size of the byte buffer in bytes units
+     * @return a newly created zeroed {@link java.nio.ByteBuffer} with the [size] bytes
+     * @see NativeBufferUtils#memoryAlloc(long)
      */
-    public static native void memorySet(final Buffer buffer, final int value, final long size); 
+    public static native ByteBuffer clearAlloc(final long size);
 
     /**
+     * Copies the value [value] into [size] byte elements starting from the [buffer] address.
      * 
-     * @param to
-     * @param from
-     * @param size
-     * @return
+     * @param buffer a {@link java.nio.ByteBuffer} to set its elements
+     * @param value an 8-bit value to set the memory blocks (buffer elements) with, starting from the buffer address
+     * @param size the number of byte elements to set
+     * @return the java-nio byte buffer for chained invocations
      */
-    public static native void memoryMove(final Buffer to, final Buffer from, final long size);
+    public static native ByteBuffer memorySet(final ByteBuffer buffer, final int value, final long size); 
 
     /**
-     * Allocates a buffer block of size = [size] bytes using {@link NativeBufferUtils#malloc(long)}, and sets its contents to zero.
+     * Manipulates the values of byte elements [size] starting from the buffers addresses, from a bytebuffer to another.
      * 
-     * @param size the size of the single memory block
-     * @return a newly created zeroed java-nio buffer with the size of [count * size]
+     * @param to the destination {@link java.nio.ByteBuffer}
+     * @param from the source {@link java.nio.ByteBuffer}
+     * @param size the number of elements to move
+     * @see NativeBufferUtils#memoryCopy(ByteBuffer, ByteBuffer, long)
      */
-    public static native Buffer clearAlloc(final long size);
+    public static native void memoryMove(final ByteBuffer to, final ByteBuffer from, final long size);
```

Comment:
Usually, `from` should come first and `to` next.

Reply:
The [native method](https://github.com/Software-Hardware-Codesign/jme-alloc/blob/28ae779618ba64293c65cfbf6c8376afe71182fe/jme3-alloc-native/src/include/jmealloc.h#L92-L96) starts with the [to] buffer, i would like to keep it aligned to the native method.

Reply:
Yeah, I meant to change both of them;)

Reply:
I mean the native method `memmove(void* to, void* from, size_t size)` from stdlib starts with the [to] buffer.

Reply:
Oh, I see. Sorry!